shell/probe/nocturnal.sh

```diff
@@ -27,4 +28,5 @@ do
             fi
         fi
     fi
+    rm -f $temp
```

Comment:
why is this in two places? we should only require 1 clean up line.

Reply:
i assume the first is just a nicety so you don't have to stare at an empty temp file while the probe sleeps. An alternative is to not use mktemp so we don't create an empty file when there are no more tests left

Reply:
@makk94 is right here, I can remove the first one but the file will stay on disk until the probe comes back from sleep

Reply:
how does raindrop compare, functionality wise?

Reply:
Raindrop uses try/finally

Reply:
I mean: how often are the temp files removed from the box?

Reply:
Right after the test cleanup

Reply:
if i'm reading raindrop right, we download and run all the tests and then we clean them all up right before sleeping, right? (we don't clean up between test runs)

in nocturnal, we clean up after every test run (and before sleeping)

Reply:
doesn't sound right... as I don't think we're storing a list of run tests in the loop, are we?

Reply:
> if i'm reading raindrop right, we download and run all the tests and then we clean them all up right before sleeping, right? (we don't clean up between test runs)
> 
> in nocturnal, we clean up after every test run (and before sleeping)

Actually we only fetch one test and sleep...

Reply:
> if i'm reading raindrop right, we download and run all the tests and then we clean them all up right before sleeping, right? (we don't clean up between test runs)
> 
> in nocturnal, we clean up after every test run (and before sleeping)

No. We fetch a single test and then sleep. Cleanup is done per test.

Reply:
we recursively call Run tho

Reply:
> we recursively call Run tho (so won't hit the finally until after all the recursions)
Updated raindrop to run after the temp file removal

Reply:
so what's the consensus?

Reply:
Updated with a single `rm`

Reply:
Another option - make this look more like demo.sh where the logic might be
1) get redirect url
2) if it doesn't have a uuid in it, sleep
3) else, follow the redirect, save the test to file, do the test, remove the file

then don't need to worry about removing a file before sleeping

Reply:
closing because this thread is long